microceph/ceph/osd.go

```diff
@@ -292,22 +295,164 @@ func setStablePath(storage *api.ResourcesStorage, param *types.DiskParameter) er
 	return nil
 }
 
+// parseBackingSpec parses a loopback file specification.
+// The specification is of the form "loop,<size><unit>,<number>".
+// The function returns the size in MB and the number of disks.
+func parseBackingSpec(spec string) (uint64, int, error) {
+	r := regexp.MustCompile("loop,([1-9][0-9]*[MGT]),([1-9][0-9]*)")
+
+	match := r.FindStringSubmatch(spec)
+	if match != nil {
```

Comment:
This is slightly different from the function structure which is (usually)"
```
if ERR {
    # RETURN ERROR
}

# PROCEED
```

but rather:
```
if SUCCESS {
    # Core Logic
}
# Return Error
```

The only nit being the structure is unconventional for quick reading.